packages/ui/src/containers/PhoneForm/use-sms-sign-in.ts

```diff
@@ -0,0 +1,89 @@
+import type { SignIn } from '@logto/schemas';
+import { SignInIdentifier } from '@logto/schemas';
+import { useState, useMemo, useCallback } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { sendSignInSmsPasscode } from '@/apis/sign-in';
+import type { ErrorHandlers } from '@/hooks/use-api';
+import useApi from '@/hooks/use-api';
+import type { ArrayElement } from '@/types';
+import { UserFlow } from '@/types';
+
+export type MethodProps = ArrayElement<SignIn['methods']>;
+
+const useEmailSignIn = ({ password, isPasswordPrimary, verificationCode }: MethodProps) => {
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const navigate = useNavigate();
+
+  const errorHandlers: ErrorHandlers = useMemo(
+    () => ({
+      'guard.invalid_input': () => {
+        setErrorMessage('invalid_phone');
+      },
+    }),
+    [setErrorMessage]
+  );
```

Comment:
This feels a little bit strange.
1. How is the `guard.invalid_input` key used in the handler?
2. This is not a `complex calculation` so it's not necessary to use this useMemo hook. `setState` method does not have to be included in the dependencies list, IMO

Reply:
1. It is a map of the server-side error codes. Its type is guarded using LogtoErrorCode. This handler hook allows customized logic for a specific given error code. 

2.1 Similar reason as above. The useApi hook treats these error handlers as a `useEffect` dependency. Have to make sure the reference won't change.

2.2 updated, can't recall why to add it, probably copy from some previous code.